File: problem-090/solution.py
from typing import Set, List
from random import randint

def random_with_exception(n: int, nums: List[int]) -> int:
    exclusion: Set[int] = set(nums)
    selection: List[int] = [num for num in range(n) if num not in exclusion]
    return selection[randint(0, len(selection))]

got = random_with_exception(10, [3, 7, 12, 9, 4])
assert got not in [3, 7, 12, 9, 4]

File: problem-090/solution.go
package problem090

import (
	"math/rand"
)

// randExcept generate random number from 0 to n excluding number in given slice.
func randExcept(n int, exclusion []int) int {
	excludes := make(map[int]bool)
	for _, num := range exclusion {
		excludes[num] = true
	}

	candidates := make([]int, 0)
	for i := 0; i <= n; i++ {
		if excludes[i] {
			continue
		}
		candidates = append(candidates, i)
	}

	if len(candidates) == 0 {
		return -1
	}

	result := candidates[rand.Intn(len(candidates)-1)]
	return result
}
